lib/datadog/core/configuration/option.rb

```diff
@@ -191,11 +189,33 @@ def coerce_env_variable(value)
           when :string, NilClass
             value
           else
-            raise ArgumentError,
+            raise InvalidDefinitionError,
               "The option #{@definition.name} is using an unsupported type option for env coercion `#{@definition.type}`"
           end
         end
 
+        def coerce_env_var_int(value)
+          # Strict integer parsing
+          Integer(value, 10)
+        rescue ArgumentError => e
+          # Try also parsing as a whole floating point numbers
+          begin
+            f = Float(value)
+
+            # Check for whole numbers
+            raise ArgumentError, "#{value} is not a whole number" unless f.truncate == f
+
+            f
+          rescue ArgumentError
+            # It's neither an integer nor a whole float
+            raise e
+          end
```

Comment:
Is it just me, or would this `rescue` swallow our `raise ArgumentError, "#{value} is not a whole number" unless f.truncate == f` above and replace it with the original `Integer` parsing error?

(I guess it may be worth adding a spec to validate that the right error/message gets presented?)